src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -79,4 +79,37 @@ public void storeMessage(Message message) {
 
     return messages;
   }
+
+  /**
+   * Gets messages posted by all users.
+   *
+   * @return a list of messages posted by all users, or empty list if no user has ever posted a
+   *     message. List is sorted by time descending.
+   */
+public List<Message> getAllMessages(){
+  List<Message> messages = new ArrayList<>();
+
+  Query query = new Query("Message")
+    .addSort("timestamp", SortDirection.DESCENDING);
+  PreparedQuery results = datastore.prepare(query);
+
+  for (Entity entity : results.asIterable()) {
+   try {
```

Comment:
How about creating a private helper method that contains this loop so that getMessages() and getAllMessages() can share this code? That way if you end up adding fields  to the messages, you only have to change the code in one spot.